packages/compass-indexes/src/modules/search-indexes.spec.ts

```diff
@@ -269,6 +273,88 @@ describe('search-indexes module', function () {
       expect(store.getState().searchIndexes.error).to.be.undefined;
     });
   });
+
+  describe('openSearchIndexes and closeSearchIndexes', function () {
+    let clock: sinon.SinonFakeTimers;
+
+    before(() => {
+      clock = sinon.useFakeTimers();
+    });
+
+    after(() => {
+      clock.restore();
+    });
+
+    it('starts and stops the polling', async function () {
+      const pollInterval = 5000;
+
+      const getSearchIndexesStub = sinon.stub().resolves(searchIndexes);
+      const store = setupStore(
+        {
+          isSearchIndexesSupported: true,
+        },
+        {
+          getSearchIndexes: getSearchIndexesStub,
+        }
+      );
+
+      const waitForStatus = async (status: FetchStatus) => {
+        await waitFor(() => {
+          return store.getState().regularIndexes.status === status;
```

Comment:
`waitFor` resolves when the function inside stops throwing errors, so it should be something like this instead

```suggestion
          expect(store.getState().regularIndexes.status).to.eq(status)
```

Reply:
Oof.  And it didn't fail because any `await` causes next tick :)